lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -192,27 +196,47 @@ def parse(body)
         JSON.parse(body)
       end
 
-      def commit(action, parameters, options = {})
+      def commit(action, parameters, options = {}, method='post')
         url = url(action, parameters, options)
         post = post_data(action, parameters)
         begin
-          raw = ssl_post(url, post, headers(post, options))
+          if method == 'post'
+            raw = ssl_post(url, post, headers(post, options))
+          else
+            raw = ssl_get(url(action, parameters, options), headers(""))
+          end
           response = parse(raw)
         rescue ResponseError => e
           raw = e.response.body
           response = parse(raw)
         end
-
-        Response.new(
-          success_from(action, response),
-          message_from(action, response),
-          response,
-          authorization: authorization_from(response),
-          avs_result: AVSResult.new(code: response['some_avs_response_key']),
-          cvv_result: CVVResult.new(response['some_cvv_response_key']),
-          test: test?,
-          error_code: error_code_from(action, response)
-        )
+        if method == 'post'
+            Response.new(
+              success_from(action, response),
+              message_from(action, response),
+              response,
+              authorization: authorization_from(response),
+              avs_result: AVSResult.new(code: response['some_avs_response_key']),
+              cvv_result: CVVResult.new(response['some_cvv_response_key']),
+              test: test?,
+              error_code: error_code_from(action, response)
+            )
+        else
+            tmp_response = response
+            if response.instance_of? Array
+              tmp_response = {'response': response}
+            end
+            Response.new(
+              success_from_get(response),
+              message_from_get(response),
+              tmp_response,
+              avs_result: nil,
+              cvv_result: nil,
+              authorization: nil,
+              test: test?,
```

Comment:
I think `message_from` will be the only difference. Can we extract the message before creating the Response depending on the HTTP_METHOD and then create Response?

Reply:
Here get method will return just list of object not any status_code or dict which says status of the call.